src/clone.rs

```diff
@@ -246,63 +244,131 @@ macro_rules! to_return_value {
 /// let closure = clone!(v => move |x| {
 ///     println!("v: {}, x: {}", v, x);
 /// });
+/// # drop(v);
+/// # closure(2);
 /// ```
 ///
 /// **Passing `self` as an argument**:
 ///
 /// ```compile_fail
 /// # use glib::clone;
 /// # use std::rc::Rc;
+/// #[derive(Debug)]
 /// struct Foo;
 ///
 /// impl Foo {
 ///     fn foo(&self) {
 ///         let v = Rc::new(1);
 ///
 ///         let closure = clone!(@strong self => move |x| {
-///             println!("v: {}, x: {}", v, x);
+///             println!("v: {:?}", self);
+///         });
+///         # drop(v);
+///         # closure(2);
+///     }
+/// }
+/// ```
+///
+/// If you want to use `self` directly, you'll need to rename it:
+///
+/// ```
+/// # use glib::clone;
+/// # use std::rc::Rc;
+/// #[derive(Debug)]
+/// struct Foo;
+///
+/// impl Foo {
+///     fn foo(&self) {
+///         let v = Rc::new(1);
+///
+///         let closure = clone!(@strong self as this => move |x| {
+///             println!("v: {:?}", this);
+///         });
+///         # drop(v);
+///         # closure(2);
+///     }
+/// }
+/// ```
+///
+/// **Passing fields directly**
+///
+/// ```compile_fail
+/// # use glib::clone;
+/// # use std::rc::Rc;
+/// #[derive(Debug)]
+/// struct Foo {
+///     v: Rc<usize>,
+/// }
+///
+/// impl Foo {
+///     fn foo(&self) {
+///         let closure = clone!(@strong self.v => move |x| {
+///             println!("v: {:?}", v);
```

Comment:
I don't think we should allow that. It allows annoying bugs to sneak in

```rust
clone!(@strong self.v, self.bla.v => move |x| {
    // what's `v` here now? Might even be the same type so the closure would silently misbehave
});
```

I believe we should require renaming for `self` and field accesses and not come up with any magic names ourselves.

Reply:
We don't. The block is marked as "compile_fail" you know. :p

Reply:
Ah please remove this from the docs then. People like me assume that examples in the docs actually work :) I didn't see that annotation

Reply:
People reading the docs will see the "this code doesn't compile" element. rustdoc stuff you know. :p

Reply:
How does that look like in the docs, can you show an example in e.g. the std docs? It seems strange to me to include broken code examples in the docs.

Reply:
Ok you convinced me